### PR TITLE
Cranelift: extend docs on Inst to discuss `call` instructions

### PR DESCRIPTION
**Pull request by iximeow**
the docs on `Inst` note that the type is returned by non-resultful
instructions built from `InstBuilder`, but did _not_ note that it is
also returned by `call` and `call_indirect`. if you're trying to learn
and use Cranelift by following the docs, this means you'd follow a doc
link to `Inst` that implies that `call` does not return a value - this
is actively misleading, since you'd want to use the returned `Inst` to
find exactly those returned values!

so, this adds a few sentences talking about the case of call `Inst`s.

### CODE REVIEW

File: cranelift/codegen/src/ir/entities.rs
Comment (iximeow) [inline cranelift/codegen/src/ir/entities.rs:99]: i'd be inclined to move this to the front of the paragraph, and leave inventive uses as "and also, here are more exotic uses", but i really don't know how often people are doing `builder.ins().call()` that they might end up here!

File: cranelift/codegen/src/ir/entities.rs
Comment (cfallin) [inline cranelift/codegen/src/ir/entities.rs:92]: Maybe we can put a note here (perhaps instead of below) to the effect of: "... like call or call-indirect instructions. To get the `Value` results of such an instruction, use `inst_results` with the returned `Inst`" ?

**Review by cfallin — APPROVED** [review]
Thanks! Happy to take as-is but there's a suggestion below in case you want to refine a bit.

File: cranelift/codegen/src/ir/entities.rs
Comment (iximeow) [reply cranelift/codegen/src/ir/entities.rs]: admittedly took me a moment to push the change but i really appreciate this suggestion! it cleans up the whole "talk about the normal thing, then the exotic thing, then the normal thing again" wrinkle nicely.

**Comment by iximeow** [comment]
i forget the etiquette here since i've pushed a change since your review, but i still have a shiny green `merge` button. CI's gotta churn anyway, so i'll let you have the last word on if this is good for a squash merge still - i assume so :D

**Review by cfallin — APPROVED** [review]
LGTM with update, thanks!

> i forget the etiquette here since i've pushed a change since your review, but i still have a shiny green merge button. CI's gotta churn anyway, so i'll let you have the last word on if this is good for a squash merge still - i assume so :D

Yup, just for the sake of norm-setting I'll note here -- I would have been totally happy to see this merge without another LGTM; I generally give an "approve with comments" when it's just little nits and I trust the PR author to clean up and merge autonomously. Because LGTMs are cheap I'll r+ again for avoidance of doubt :-)

**Pull request merged**
